# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/gc/GC.cpp
# Commit: 1937b90d16a4
# Full Hash: 1937b90d16a4714abbd46473d5b9ff76c4435ed8
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2020-07-22 04:32:41
# Description:
#   Bug 1654186 - Call onOutOfMallocMemory() on OOM in GCRuntime::decommitFreeArenas so that we release protected free areans before trying to decommit them r=sfink
#   
#   Compacting temporarily mprotects released arenas in debug builds to catch problems. This interferes with Chunk::decommitFreeArenasWithoutUnlocking which checks for free arenas by looking at every arena in the chunk. This is different to Chunk::decommitOneFreeArena which uses the list of free arenas.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D84334
# ==============================================================================

diff -r 654c839a370a -r 1937b90d16a4 js/src/gc/GC.cpp
--- a/js/src/gc/GC.cpp	Mon Jul 20 22:20:17 2020 +0000
+++ b/js/src/gc/GC.cpp	Tue Jul 21 15:27:41 2020 +0000
@@ -3254,7 +3254,7 @@
        chunk.next()) {
     if (chunk->info.numArenasFreeCommitted != 0 &&
         !chunksToDecommit.append(chunk)) {
-      decommitFreeArenasWithoutUnlocking(lock);
+      onOutOfMallocMemory(lock);
       return;
     }
   }